Search sidebar: scroll to header

KarinaMendez2 opened the pull request.

Commit:
first attempt to get to the top
of search sidebar on every
new search

Commit:
unit test for visibility in search
sidebar header

Commit:
adding refs to the test

Commit:
added recieveSearchResults for SearchResultsBarWrapper
to be rendered in tree

Commit:
Merge remote-tracking branch 'origin/master' into search_sidebar_header

TomWoodward commented on src/app/content/search/components/SearchResultsSidebar/SearchResultsBarWrapper.tsx:
```diff
@@ -38,7 +39,9 @@ export class SearchResultsBarWrapper extends Component<ResultsSidebarProps> {
-  public totalResults = () => <Styled.SearchQueryWrapper>
+  public totalResults = () => <Styled.SearchQueryWrapper
+    ref={this.searchSidebarHeader}
+    data-testid='search-results-sidebar-header'>
```
you don't need this testid anymore right?

TomWoodward commented on src/app/content/search/components/SearchResultsSidebar/index.spec.tsx:
```diff
@@ -47,7 +50,7 @@ describe('SearchResultsSidebar', () => {
     <MessageProvider>
       <Provider store={store}>
-        <SearchResultsSidebar />
+        <SearchResultsSidebar></SearchResultsSidebar>
```
can revert this right?

TomWoodward commented on src/app/content/search/components/SearchResultsSidebar/index.spec.tsx:
```diff
@@ -175,4 +178,30 @@ describe('SearchResultsSidebar', () => {
+    store.dispatch(requestSearch('cool search'));
+    store.dispatch(
+      receiveSearchResults(
+        makeSearchResults([makeSearchResultHit({ book: archiveBook, page })])
```
if you don't care what the results are you don't have to pass anything to the `makeSearchResults` helper

TomWoodward commented on src/app/content/search/components/SearchResultsSidebar/index.spec.tsx:
```diff
@@ -175,4 +178,30 @@ describe('SearchResultsSidebar', () => {
+    const header = sidebar.searchSidebarHeader.current;
+    const scrollSidebarSectionIntoViewMock = jest.spyOn(domUtils, 'scrollSidebarSectionIntoView');
+
+    expect(scrollSidebarSectionIntoViewMock).not.toHaveBeenCalled();
```
since you just made this spy, it definitely won't have been called yet, you can probably remove this assertion

TomWoodward commented on src/app/content/search/components/SearchResultsSidebar/index.spec.tsx:
```diff
@@ -175,4 +178,30 @@ describe('SearchResultsSidebar', () => {
+    store.dispatch(requestSearch('second cool search'));
+    store.dispatch(
+      receiveSearchResults(
+        makeSearchResults([makeSearchResultHit({ book: archiveBook, page })])
```
same as above

Commit:
Merge remote-tracking branch 'origin/master' into search_sidebar_header

Commit:
removed assertion, not needed
params and reverted change
in tree for test, removed data
test id in SearchResultsBarWrapper

Commit:
updated snap
